compiler-rs/openapi_to_clients_schema/src/main.rs

```diff
@@ -26,31 +26,40 @@ fn main() -> anyhow::Result<()> {
     let subscriber = FmtSubscriber::builder().with_max_level(Level::TRACE).finish();
     tracing::subscriber::set_global_default(subscriber)?;
 
-    // let file = "../../ent-search/swagger/v1/enterprise-search.json";
-    // let file = "../../ent-search/swagger/v1/workplace-search.json";
-    let file = "./openapi_to_clients_schema/fixtures/workplace-search.json";
+    // let path = "../../../ent-search/swagger/enterprise-search.json";
+    let path = "./fixtures/kibana.serverless.yaml";
 
-    let src = PathBuf::from(file);
-    let dest = src.with_extension("schema.json");
-
-    info!("Loading OpenAPI from {file}");
-
-    let file = std::fs::File::open(file)?;
-
-    let json_deser = &mut serde_json::Deserializer::from_reader(file);
+    info!("Loading OpenAPI from {path}");
+    let file = std::fs::File::open(path)?;
 
     // Track unused fields, to find any additional stuff the OpenAPI model would miss
     let mut unused = BTreeSet::new();
 
-    let open_api: openapiv3::OpenAPI = serde_ignored::deserialize(json_deser, |path| {
-        unused.insert(path.to_string());
-    })?;
+    let open_api = match Path::new(path).extension() {
+        Some(ext) if ext == "json" => {
+            let deser = &mut serde_json::Deserializer::from_reader(file);
+            serde_ignored::deserialize(deser, |path| {
+                unused.insert(path.to_string());
+            })
```

Comment:
Unfortunately `serde_json::Deserializer::from_reader()` is &mut, while `serde_yml::Deserializer::from_reader(file)` is not, which as far as I can tell forces duplicating the `deserialize()` call.

Reply:
I would change the logic to read the file into a `String` and change the code to:

```suggestion
            let mut deser = serde_json::Deserializer::from_str(&data);
            serde_ignored::deserialize(&mut deser, |path| {
                unused.insert(path.to_string());
            })
```

Given that data comes from:
```rust
    let mut data = String::new();
    std::fs::File::open(path)?.read_to_string(&mut data)?;
```

Counter intuitive stuff happens with reader and buffers. [from_reader doc](https://docs.rs/serde_json/latest/serde_json/de/fn.from_reader.html)

Reply:
Thanks, done in [`17ca650` (#3195)](https://github.com/elastic/elasticsearch-specification/pull/3195/commits/17ca650923eb24e31a620c5b1d86e00617fd9997). I did not realize using a string would be much faster.